css-transitions-2/Overview.bs

```diff
@@ -120,6 +122,42 @@ the reversing shortening factor.
 The 'transition-delay' property specifies the [=start delay=]
 of the transition's associated [=animation effect=].
 
+## The 'transition-behavior' Property ## {#transition-behavior-property}
+
+The 'transition-behavior' property specifies whether transitions will be
+started or not for discrete properties.
+
+<pre class="propdef">
+  Name: transition-behavior
+  Value: normal | allow-discrete
+  Initial: normal
+  Applies to: All elements
+  Inherited: no
+  Percentages: N/A
+  Computed value: as specified
+  Canonical order: per grammar
+  Animation type: not animatable
+</pre>
+
+The syntax for specifying 'transition-behavior' is as follows:
+
+<div class="prod">
+  <dfn type id="transition-behavior-value">&lt;transition-behavior-value&gt;</dfn> = [ 'normal' | 'allow-discrete' ]
+</div>
+
+When 'normal' is specified, animations will not be started for discrete
+properties, only for interpolable properties. When 'allow-discrete' is
+specified, animations will be started for discrete properties as well as
+interpolable properties.
+
+## The 'transition' Shorthand Property ## {#transition-shorthand-property}
+
+The syntax for specifying 'transition' is as follows:
+
+<div class="prod">
+  <dfn type id="single-transition">&lt;single-transition&gt;</dfn> = <<transition-behavior-value>> || [ ''transition-property/none'' | <<single-transition-property>> ] || <<time>> || <<easing-function>> || <<time>>
```

Comment:
I think it's probably best to put the `<<transition-behavior-value>>` last; the order does affect the order of serialization that the spec requires.

Reply:
We might want to leave the normal value out of the shorthand - it is the initial value so unnecessary to be specified, and it would prevents its specification in other longhands in the future.

Reply:
I think leaving `normal` out of the shorthand is an interesting idea -- but I also think it's not something the WG has done before for any shorthand, so if we think it's worth doing, I think we should bring it up explicitly to the WG to see if other folks have feedback on the idea.

Reply:
I realize that `transition-property` takes `<custom-ident>` values, which means that the spec should probably also have wording excluding the two new keywords from that `<custom-ident>` as described in https://drafts.csswg.org/css-values-4/#identifier-value , although that change does have (low, I think) compatibility risk.

Or, alternatively, we could choose *not* to follow that advice and exclude the keywords, in which case we probably *do* want to leave this in the first position.

Reply:
> I think it's probably best to put the <<transition-behavior-value>> last; the order does affect the order of serialization that the spec requires.

So instead of doing `transition: allow-discrete display 2s` it should be `transition: display 2s allow-discrete`? Now that the value is called "allow-discrete" I'm more ok with this. When it was just "discrete" or "animatable" I felt more strongly that it should come first. What do you think rob?

> the spec should probably also have wording excluding the two new keywords from that `<custom-ident>`

If we put the keyword at the end, then this wouldn't be needed right?

Reply:
I don't think the ordering is relevant.  This affects the parsing rules -- and effectively without that exclusion the spec would need prose rules for how to distinguish the values, given that https://drafts.csswg.org/css-values-4/#component-combinators explicitly says that the ordering around `||` does *not* matter.  (This implies, I think, that it doesn't imply any rule about breaking ties during parsing when more than one option of an `||` matches.  Though maybe that implication wasn't intended.  But I think it does imply that it's a specification error if more than one option around an `||` matches and the spec doesn't say how to resolve the conflict.)

Reply:
> given that https://drafts.csswg.org/css-values-4/#component-combinators explicitly says that the ordering around || does not matter

Oh wow, so `transition: 2s color` works the same as `transition: color 2s`!? It looks like this is true after I tried a basic test in chrome.

I moved it to the end of the list